File: server/src/models/medicine.js
const mongoose = require('mongoose');

const Schema = mongoose.Schema;

const MedicineSchema = new Schema ({
    medicineId : { type : Number, required : true, unique : true },
    name : { type : String, required : true },
    company : { type : String, required : true },
    target : { type : String, required : true },
    dosageInfo : { type : String, required : true },
    warn : { type : String, required : true },
    antiEffect : { type : String, required : true }
})

MedicineSchema.statics.findByKeyword = function(keyword) {
    return this.find({
        $or : [
            { name : { $regex : keyword }},
            { company : { $regex : keyword }},
            { target : { $regex : keyword }},
        ]
    })
};

MedicineSchema.statics.findByMedicineId = function(medicineId) {
    return this.findOne({ medicineId })
};


module.exports = mongoose.model('Medicine', MedicineSchema);

File: app/flutter_application_1/lib/src/screens/models/Medicine.dart
class Medicine {
  final int medicineId;
  final String antiEffect;
  final String company;
  final String dosage;
  final String name;
  final String target;
  final String warn;

  Medicine(
      {this.medicineId,
      this.antiEffect,
      this.company,
      this.dosage,
      this.name,
      this.target,
      this.warn});

  factory Medicine.fromJson(Map<String, dynamic> parsedJson) {
    return Medicine(
      medicineId: parsedJson['medicineId'],
      antiEffect: parsedJson['antiEffect'],
      company: parsedJson['company'],
      dosage: parsedJson['dosage'],
      name: parsedJson['name'],
      target: parsedJson['target'],
      warn: parsedJson['warn'],
    );
  }

  Map<String, dynamic> toJson() => {
        "medicineId": medicineId,
        "antiEffect": antiEffect,
        "company": company,
        "dosage": dosage,
        "name": name,
        "target": target,
        "warn": warn,
      };
}
